scripts/build-single-packages.js

```diff
@@ -19,27 +20,42 @@ if (!configJson.modules || configJson.modules.length === 0) {
   process.exit(1);
 }
 
-const components = configJson.modules.map(module => ({
+const components = configJson.modules.map(({ name, type, moduleGlob }) => ({
+  // need the /*/*/ to avoid grabbing top level index files
+  /**
+   * We don't want the /index.js or /components/index.js to be have packages
+   * These files will not help with tree shaking in module federation environments
+   */
     files: glob
-      .sync(`${root}/${module.name}/**/**/index.js`)
+      .sync(moduleGlob ? `${root}${moduleGlob}` : `${root}/${name}/*/*/**/index.js`)
       .filter((item) => !foldersexclude.some((name) => item.includes(name)))
       .map((name) => name.replace(/\/$/, '')),
-    type: module.type
-  }));
+    type
+  }))
+
+
 
 async function createPackage(component, dist) {
   const cmds = [];
-  const destFile = component.replace(/index.js/g, 'package.json');
+  let destFile = component.replace(/[^/]+\.js$/g, 'package.json').replace('/dist/esm/', '/dist/dynamic/');
+  if(component.match(/index\.js$/)) {
+    destFile = component.replace(/[^/]+\.js$/g, 'package.json').replace('/dist/esm/', '/dist/dynamic/');
+  } else {
+    destFile = component.replace(/\.js$/g, '/package.json').replace('/dist/esm/', '/dist/dynamic/');
+  }
   const pathAsArray = component.split('/');
-  const esmRelative = `.${component.split(`${root}`)[1]}`;
+  const destDir = destFile.replace(/package\.json$/, '')
+  const esmRelative = path.relative(destDir, component)
+  const cjsRelative = path.relative(destDir, component.replace('/dist/esm/', '/dist/js/'))
+  const typesRelative = path.relative(destDir, component.replace(/\.js$/, '.d.ts'))
 
   const packageName = configJson.packageName;
   if (!packageName) {
     console.log("packageName is required!")
     process.exit(1);
   }
 
-  let componentName = pathAsArray[pathAsArray.length - 2];
+  let componentName = pathAsArray[pathAsArray.length - (component.match(/index\.js$/) ? 2 : 1)];
   if (pathAsArray.includes("next")) {
     componentName = `${componentName.toLowerCase()}-next-${dist}`;
```

Comment:
Thinking we should change this to dynamaic instead of the dist type since we only have one package.json with a main and a module in it.  thoughts?